lib/rule-list.js

```diff
@@ -155,17 +155,95 @@ export const SIMPLIFY_FRACTION = defineRuleString(
 // e.g. 2/-3 -> -2/3
 export const SIMPLIFY_SIGNS = defineRuleString('#a / -#b', '-#a / #b')
 
+// e.g. 2x/3 -> 2/3 x
+export const REWRITE_FRACTIONAL_POLYNOMIAL = defineRuleString(
+    '#a #b / #c', '#a / #c #b', {a: query.isNumber, b: isPolynomialTerm, c: query.isNumber}
+)
 
 // ADDING FRACTIONS
 
+// e.g. 2/5 + 2/5 -> (2 + 2)/5
 export const COMBINE_NUMERATORS =
     defineRuleString('#a_0 / #b + ...', '(#a_0 + ...) / #b')
 
-export const COMMON_DENOMINATOR =
-    defineRuleString(
-        '#a_0 / #b_0 + ...',
-        '(#a_0 * #eval(lcm(#b_0, ...) / #b_0)) / (#b_0 * #eval(lcm(#b_0, ...) / #b_0)) + ...'
-    )
+// Finds common denominator in any scenario
+
+// e.g. 2/6 + 1/4 -> (2 * 2) / (6 * 2) + (1 * 3) / (4 * 3)
+// e.g. 2/(3 + x) + 2/3 -> (2 * 3)/ (3 * (3 + x)) + 2 * (3 + x) / (3 * (3 + x))
+export const COMMON_DENOMINATOR = defineRule(
+    (node) => {
+        if (query.isAdd(node)) {
+            const terms = node.args
+            // checks if at least one of the terms is a fraction
+            const hasFraction = terms.some(term => query.isDiv(term))
+
+            let sameDenom
+            // rule cannot apply if all of the terms are fractions
+            // and they all have the same denominator
+            if (terms.every(term => query.isDiv(term))) {
+                const denom = terms[0].args[1]
+                sameDenom = terms.every(term => print(term.args[1]) == print(denom))
+            }
+
+            return hasFraction && !sameDenom ? {node} : null
+        }
+    },
+
+    (node) => {
+        const terms = node.args
+        // an array storing the index where fraction is negative
+        const negatives = terms.map(term => query.isNeg(term))
```

Comment:
It would be could if we could extract this logic for reuse.  Don't worry about it for this review, but something to think about.